GiEnJul/ClientApp/src/common/components/NavMenuAdmin.tsx

```diff
@@ -1,43 +1,90 @@
-                            <ul className="navbar-nav flex-grow">
-                                <NavItem>
-                                    <NavLink tag={Link} className="text-dark" to="/admin">Admin</NavLink>
-                                </NavItem>
-                                <NavItem>
-                                    <NavLink tag={Link} className="text-dark" to="/registrer-familie">Registrer familie</NavLink>
-                                </NavItem>
-                                <NavItem>
-                                    <NavLink tag={Link} className="text-dark" to="/rediger">Rediger nettside</NavLink>
-                                </NavItem>
-                            </ul>
-                        </Collapse>
-                    </Container>
-                </Navbar>
-        );
+  public render() {
+    if (this.props.role == "Institution") {
+      return (
+        <Navbar
+          className="navbar-expand-sm navbar-toggleable-sm border-bottom box-shadow"
+          light
+        >
+          <Container>
+            <NavbarBrand tag={Link} to="/">
+              <img className={"logo-small"} src={logo}></img>
+            </NavbarBrand>
+            <NavbarToggler onClick={this.toggle} className="mr-2" />
+            <LogOutButton />
+          </Container>
+        </Navbar>
+      );
+    } else {
+      return (
+        <Navbar
+          className="navbar-expand-sm navbar-toggleable-sm border-bottom box-shadow"
+          light
+        >
+          <Container>
+            <NavbarBrand tag={Link} to="/">
+              <img className={"logo-small"} src={logo}></img>
+            </NavbarBrand>
+            <NavbarToggler onClick={this.toggle} className="mr-2" />
+            <Collapse
+              className="d-sm-inline-flex flex-sm-row-reverse"
+              isOpen={this.state.isOpen}
+              navbar
+            >
+              <ul className="navbar-nav flex-grow">
+                <NavItem>
+                  <NavLink tag={Link} className="text-dark" to="/admin">
+                    Admin
+                  </NavLink>
+                </NavItem>
+                <NavItem>
+                  <NavLink
+                    tag={Link}
+                    className="text-dark"
+                    to="/registrer-familie"
+                  >
+                    Registrer familie
+                  </NavLink>
+                </NavItem>
+                <NavItem>
+                  <NavLink tag={Link} className="text-dark" to="/rediger">
+                    Rediger nettside
+                  </NavLink>
+                </NavItem>
+              </ul>
+            </Collapse>
+            <LogOutButton />
+          </Container>
+        </Navbar>
+      );
     }
+  }
 
-    private toggle = () => {
-        this.setState({
-            isOpen: !this.state.isOpen
-        });
-    }
+  private toggle = () => {
+    this.setState({
+      isOpen: !this.state.isOpen,
+    });
```

Comment:
I think the prev state should be used here to avoid state bugs.
```suggestion
    this.setState((prev) => ({
      isOpen: !prev.isOpen,
    }));
```

Reply:
Did not think too much about how the navmenuadmin.tsx file looked cause of issue #301. Didn't want the PR to get too big

Reply:
Oh I agree. This should probably be done in #301 then.

Reply:
Hmm yeah it probably does.